1단계 - 학습테스트

Ppojin opened the pull request:
첫 PR 입니당 🎉
![image](https://github.com/g-oop/goop-java-racingcar/assets/25076967/1252ceca-150e-4008-b6c6-3a6d47e49700)

Ppojin committed:
test: StringUtil

Ppojin committed:
feature: add StringUtils

Ppojin committed:
test: 테스트 이름 변경

Ppojin committed:
feature: substring 이용

요구사항 2

Ppojin committed:
test: 생명주기 어노테이션 제거

Ppojin committed:
test: Set Collection

Ppojin committed:
test: 변수명 수정

Ppojin committed:
fix: 괄호 제거 조건

yamsroun commented:
꼼꼼이 신경써서 잘 작성해 주셨네요~ 짱짱맨~
IntelliJ의 기능을 최대한 활용해서 아래 2건만 조치해 주시면 좋을 것 같아요~
- 불필요한 import문 제거
- 유틸리티 클래스에 `private` 생성자 추가
<img width="830" alt="image" src="https://github.com/g-oop/goop-java-racingcar/assets/20655591/8e6461b5-9779-48c2-ac1a-0149458e88c2">

yamsroun commented:
사소하지만, 요것도 IntelliJ의 Inspector가 알려주네요~
<img width="1194" alt="image" src="https://github.com/g-oop/goop-java-racingcar/assets/20655591/52f3830a-cd20-4d9d-b104-0fa532099f0d">
<img width="1248" alt="image" src="https://github.com/g-oop/goop-java-racingcar/assets/20655591/6e8fc639-2886-4b9f-a19c-b87cf45a94a4">

yamsroun commented:
`SetTest` 클래스의 2, 3번 테스트는 `containsExactly()`를 이용하면 하나의 TC로 가능해 보이네요.
```java
Assertions.assertThat(numbers).containsExactly(test);
```

yamsroun commented:
바로 위 코멘트는 요구사항을 보지 않고 달았네요. 😅

Ppojin commented:
> 꼼꼼이 신경써서 잘 작성해 주셨네요~ 짱짱맨~ IntelliJ의 기능을 최대한 활용해서 아래 2건만 조치해 주시면 좋을 것 같아요~
> 
> * 불필요한 import문 제거
> * 유틸리티 클래스에 `private` 생성자 추가
> 
> <img alt="image" width="830" src="https://private-user-images.githubusercontent.com/20655591/295238453-8e6461b5-9779-48c2-ac1a-0149458e88c2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ4Nzc1MjIsIm5iZiI6MTcwNDg3NzIyMiwicGF0aCI6Ii8yMDY1NTU5MS8yOTUyMzg0NTMtOGU2NDYxYjUtOTc3OS00OGMyLWFjMWEtMDE0OTQ1OGU4OGMyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTEwVDA5MDAyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTExNTdlYTBmMzk4OTJiYzMxNDlkNDk5NDQ3MGEwNzM1NjFmYTg5ZTEwNTAzNjMxNjk5NThjODM1NDZjNTBjZmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Jlz7pQIR9d3iQ7TYzbrEGigwZ4fniLP7cKJTIlT3g9A">


> 사소하지만, 요것도 IntelliJ의 Inspector가 알려주네요~ <img alt="image" width="1194" src="https://private-user-images.githubusercontent.com/20655591/295239965-52f3830a-cd20-4d9d-b104-0fa532099f0d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ4Nzc1MjIsIm5iZiI6MTcwNDg3NzIyMiwicGF0aCI6Ii8yMDY1NTU5MS8yOTUyMzk5NjUtNTJmMzgzMGEtY2QyMC00ZDlkLWIxMDQtMGZhNTMyMDk5ZjBkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTEwVDA5MDAyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ0MTA1YmRlMjc0ZGUzMzAyZjIzZGM2NTYwMWVkMjI4MmUxY2Y4NGIyMzA4YjhlNGIyZTRlZTRiOTU4YmQ0M2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fALa7AjxwLxZuE3ighSOLWfi6ZbjyDPLf_TJlb3DWe4"> <img alt="image" width="1248" src="https://private-user-images.githubusercontent.com/20655591/295240096-6e8fc639-2886-4b9f-a19c-b87cf45a94a4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ4Nzc1MjIsIm5iZiI6MTcwNDg3NzIyMiwicGF0aCI6Ii8yMDY1NTU5MS8yOTUyNDAwOTYtNmU4ZmM2MzktMjg4Ni00YjlmLWExOWMtYjg3Y2Y0NWE5NGE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTEwVDA5MDAyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ1MTFkZjM1YzY5ZGViYzBjMGNjNDQ2MGYzNzZlMDNhNjc3YjczY2E4NDQyMzFhZGJmOGFiYTY0OTExMDA3OGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6czzElI0zjvQiDLLVK0r680s4X8KQ6SGO-l948d2mlU">

오 감사합니다 @yamsroun 님과 같은 제안은 원래는 안떴는데 SonarLInt 플러그인 설치하니까 저도 보이네요 ㅎㅎ

---

> `SetTest` 클래스의 2, 3번 테스트는 `containsExactly()`를 이용하면 하나의 TC로 가능해 보이네요.
> 
> ```java
> Assertions.assertThat(numbers).containsExactly(test);
> ```

> 바로 위 코멘트는 요구사항을 보지 않고 달았네요. 😅

ㅋㅋ 참고하겠습니다 감사합니다~

Ppojin committed:
refactor: StringUtils class

- 미사용 import 삭제
- private 생성자 추가

Ppojin committed:
refactor: SetTest class

- static import Assertions#assertThat
- SetTest 가시성 `public -> package-private` (JUnit 5)
- 기타 Assertion method 가독성 향상

Ppojin commented on src/main/java/StringUtils.java:
```diff
@@ -1,8 +1,8 @@
-import java.util.Set;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
-
```
미사용 import 삭제

Ppojin commented on src/main/java/StringUtils.java:
```diff
@@ -1,8 +1,8 @@
+
+    private StringUtils() {
+    }
+
```
private 생성자 추가

Ppojin commented:
리뷰해 주신 건 조치해서 커밋 올렸습니다~
